telegram/dispatcher.py

```diff
@@ -254,21 +257,23 @@ def processUpdate(self, update, context=None):
             handled = True
 
         # Telegram update (regex)
-        if isinstance(update, Update):
+        if isinstance(update, Update) and update.message is not None:
             self.dispatchRegex(update, context)
             handled = True
 
-        # Telegram update (command)
-        if isinstance(update, Update) \
-                and update.message.text.startswith('/'):
-            self.dispatchTelegramCommand(update, context)
-            handled = True
-
-        # Telegram update (message)
-        elif isinstance(update, Update):
-            self.dispatchTelegramMessage(update, context)
-            handled = True
+            # Telegram update (command)
+            if update.message.text.startswith('/'):
+                self.dispatchTelegramCommand(update, context)
 
+            # Telegram update (message)
+            else:
+                self.dispatchTelegramMessage(update, context)
+                handled = True
+        elif isinstance(update, Update) and \
+                (update.inline_query is not None or
+                 update.chosen_inline_result is not None):
+                self.dispatchTelegramInline(update, context)
+                handled = True
```

Comment:
I think that this method (`processUpdate()`)  became too complex and hard to follow the flow (fwiw, it wasn't simple before neither :-\ )
I suggest refactoring it and possibly splitting it into sub-methods to handle the internal flows.